archive_repo.py

```diff
@@ -0,0 +1,667 @@
+    + gql_AuthorFields
+)
+
+# Pull Requests
+
+gql_Review_Fields = (
+    """
+fragment reviewFields on PullRequestReview {
+    id
+    commit { abbreviatedOid }
+    ...author
+    state
+    ...commentFields
+    comments(first: 50) {
+        nodes {
+            originalPosition
+            ...commentFields
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+}
+"""
+    + gql_Comment_Fields
+    + gql_AuthorFields
+)
+
+gql_PullRequest_Fields = (
+    """
+fragment prFields on PullRequest {
+    number
+    id
+    title
+    url
+    state
+    ...labels
+    ...author
+    ...commentFields
+    closedAt
+    mergedAt
+    mergedBy { ...userFields }
+    comments(first: 100) {
+        nodes {
+            ...author
+            ...commentFields }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+    reviews(first: 50) {
+        nodes {
+            ...reviewFields
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+}
+"""
+    + gql_LabelFields
+    + gql_Review_Fields
+)
+# ...reviewFields definition includes ...commentFields and ...author
+# ...author definition includes ...userFields
+
+gql_PullRequest_Query = (
+    """
+      nodes { ...prFields }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+      totalCount
+    }
+  }
+  rateLimit {
+    remaining
+    resetAt
+  }
+}
+"""
+    + gql_PullRequest_Fields
+)
+
+gql_AllPRs_Initial = (
+    """
+query($owner: String!, $repo: String!){
+  repository(owner: $owner, name: $repo) {
+    pullRequests(first: 10, orderBy: {field: UPDATED_AT, direction:DESC}) {
+"""
+    + gql_PullRequest_Query
+)
+
+gql_AllPRs_Subsequent = (
+    """
+query($owner: String!, $repo: String!, $cursor: String!){
+  repository(owner: $owner, name: $repo) {
+    pullRequests(first: 25, after: $cursor, orderBy: {field: UPDATED_AT, direction:DESC}) {
+"""
+    + gql_PullRequest_Query
+)
+
+gql_PR_Comments_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequest {
+            comments(first:100, after:$cursor) {
+                nodes { ...commentFields }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Comment_Fields
+)
+
+gql_PR_Review_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequest {
+            reviews(first:100, after:$cursor) {
+                nodes { ...reviewFields }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
```

Comment:
It would be awesome if this could be added to every query automatically, but if it can't then a variable is probably better.

Reply:
I'll see if it's possible to pull into a fragment.